mne/io/edf/tests/test_edf.py

```diff
@@ -193,8 +193,53 @@ def test_bdf_crop_save_stim_channel(tmp_path):
 def test_edf_others(fname, stim_channel):
     """Test EDF with various sampling rates and overlapping annotations."""
     _test_raw_reader(
-        read_raw_edf, input_fname=fname, stim_channel=stim_channel, verbose="error"
+        read_raw_edf,
+        input_fname=fname,
+        stim_channel=stim_channel,
+        verbose="error",
+        test_preloading=False,
+        preload=True,  # no preload=False for mixed sfreqs
+    )
```

Comment:
Can we add a second run of this with something like:
```
with pytest.warns(...):
    _test_raw_reader(...)
```
? In theory it should still pass, right?

Reply:
No, the problem is that the `_test_raw_reader` also tests for loading slices without preloading vs loading slices with preloading. These will have different edge artifacts and therefore will fail the numpy testing. Therefore `_test_raw_reader` will not only throw a warning